pkg/agent/flowexporter/flowrecords/flow_records.go

```diff
@@ -37,11 +36,12 @@ func NewFlowRecords() *FlowRecords {
 
 // AddOrUpdateFlowRecord adds or updates the flow record in the record map given the connection.
 // It makes a copy of the connection object to record, to avoid race conditions between the
-// connection store and the flow exporter.
+// connection store and the flow exporter. We expect caller to hold the lock for
+// the connection store.
 func (fr *FlowRecords) AddOrUpdateFlowRecord(key flowexporter.ConnectionKey, conn *flowexporter.Connection) error {
-	// If the connection is in dying state and the corresponding flow records are already
-	// exported, then there is no need to add or update the record.
-	if flowexporter.IsConnectionDying(conn) && conn.DoneExport {
```

Comment:
>Do we need to check whether connection is dying here again?

Changed the name of the flag. Removed this extra check here and other places as well.